server/src/main/java/org/elasticsearch/http/DefaultRestChannel.java

```diff
@@ -101,25 +101,29 @@ public void sendResponse(RestResponse restResponse) {
         String opaque = null;
         String contentLength = null;
 
+        boolean isHeadRequest = false;
         try {
-            final BytesReference content = restResponse.content();
-            if (content instanceof Releasable) {
-                toClose.add((Releasable) content);
+            if (request.method() == RestRequest.Method.HEAD) {
+                isHeadRequest = true;
             }
-            toClose.add(this::releaseOutputBuffer);
-
-            BytesReference finalContent = content;
-            try {
-                if (request.method() == RestRequest.Method.HEAD) {
-                    finalContent = BytesArray.EMPTY;
+        } catch (IllegalArgumentException ignored) {
+            assert restResponse.status() == RestStatus.METHOD_NOT_ALLOWED
+                : "request HTTP method is unsupported but HTTP status is not METHOD_NOT_ALLOWED(405)";
+        }
+        try {
+            final HttpResponse httpResponse;
+            if (isHeadRequest == false && restResponse.isChunked()) {
+                httpResponse = httpRequest.createResponse(restResponse.status(), restResponse.chunkedContent());
+            } else {
+                final BytesReference content = restResponse.content();
```

Comment:
No this will currently just return `null` for the content, which will then be ignored below transparently (because for head we replace it with `BytesArray.EMPTY` and only do an `instanceof` check on the variable otherwise).

Reply:
ah, ofc, thanks.